src/Modules/RoundRankingModule/Services/RoundRankingService.cs

```diff
@@ -0,0 +1,231 @@
+﻿using EvoSC.Common.Interfaces;
+using EvoSC.Common.Interfaces.Models;
+using EvoSC.Common.Interfaces.Services;
+using EvoSC.Common.Interfaces.Themes;
+using EvoSC.Common.Services.Attributes;
+using EvoSC.Common.Services.Models;
+using EvoSC.Common.Util.MatchSettings;
+using EvoSC.Manialinks.Interfaces;
+using EvoSC.Modules.Official.RoundRankingModule.Config;
+using EvoSC.Modules.Official.RoundRankingModule.Interfaces;
+using EvoSC.Modules.Official.RoundRankingModule.Models;
+
+namespace EvoSC.Modules.Official.RoundRankingModule.Services;
+
+[Service(LifeStyle = ServiceLifeStyle.Singleton)]
+public class RoundRankingService(
```

Comment:
After looking at this entire class. I would split this up and only include necessary methods for updating/getting the state variables in this service. And then put methods that does some specialized operation on that data in a transient/scoped service. This way we don't have a do-it-all singleton that is references everywhere in memory and keep the single responsibility principle intact.

By doing this, it is way easier to manage the thread-safe operations on these variables as well.